pkg/verify/tsa.go

```diff
@@ -98,39 +84,16 @@ func VerifyTimestampAuthorityWithThreshold(entity SignedEntity, trustedMaterial
 	return verifiedTimestamps, nil
 }
 
-func verifySignedTimestamp(signedTimestamp []byte, dsseSignatureBytes []byte, trustedMaterial root.TrustedMaterial, verificationContent VerificationContent) (Timestamp, error) {
-	certAuthorities := trustedMaterial.TimestampingAuthorities()
+func verifySignedTimestamp(signedTimestamp []byte, dsseSignatureBytes []byte, trustedMaterial root.TrustedMaterial) (*root.Timestamp, error) {
+	timestampAuthorities := trustedMaterial.TimestampingAuthorities()
 
 	// Iterate through TSA certificate authorities to find one that verifies
-	for _, ca := range certAuthorities {
-		trustedRootVerificationOptions := tsaverification.VerifyOpts{
-			Roots:          []*x509.Certificate{ca.Root},
-			Intermediates:  ca.Intermediates,
-			TSACertificate: ca.Leaf,
-		}
-
-		// Ensure timestamp responses are from trusted sources
-		timestamp, err := tsaverification.VerifyTimestampResponse(signedTimestamp, bytes.NewReader(dsseSignatureBytes), trustedRootVerificationOptions)
-		if err != nil {
-			continue
-		}
-
-		if !ca.ValidityPeriodStart.IsZero() && timestamp.Time.Before(ca.ValidityPeriodStart) {
-			continue
-		}
-		if !ca.ValidityPeriodEnd.IsZero() && timestamp.Time.After(ca.ValidityPeriodEnd) {
-			continue
+	for _, tsa := range timestampAuthorities {
+		ts, err := tsa.Verify(signedTimestamp, dsseSignatureBytes)
+		if err == nil {
+			return ts, nil
 		}
-
-		// Check tlog entry time against bundle certificates
-		// TODO: technically no longer needed since we check the cert validity period in the main Verify loop
-		if !verificationContent.ValidAtTime(timestamp.Time, trustedMaterial) {
```

Comment:
Just wanted to check, we removed this - this was intentional? I agree that I think it's fine to remove as we should decouple certificate and timestamp verification.

Reply:
Yes, this was removed because the validity period check occurs outside this function now. 👍🏻